src/main/java/org/broadinstitute/hellbender/utils/config/GATKConfig.java

```diff
@@ -0,0 +1,135 @@
+    // General Options:
+    // ----------------------------------------------------------
+
+    @Key("system.gatk_stacktrace_on_user_exception")
+    @DefaultValue("true")
+    boolean gatk_stacktrace_on_user_exception();
+
+    // ----------------------------------------------------------
+    // SAMJDK Options:
+    // ----------------------------------------------------------
+
+    @Key("system.samjdk.use_async_io_read_samtools")
+    @ConverterClass(CustomBooleanConverter.class)
+    @DefaultValue("false")
+    Boolean samjdk_use_async_io_read_samtools();
+
+    @Key("system.samjdk.use_async_io_write_samtools")
+    @DefaultValue("true")
+    boolean samjdk_use_async_io_write_samtools();
+
+    @Key("system.samjdk.use_async_io_write_tribble")
+    @DefaultValue("false")
+    boolean samjdk_use_async_io_write_tribble();
+
+    @Key("system.samjdk.compression_level")
+    @DefaultValue("1")
+    int samjdk_compression_level();
+
+    // ----------------------------------------------------------
+    // Spark Options:
+    // ----------------------------------------------------------
+
+    @Key("system.spark.kryoserializer.buffer.max")
+    @DefaultValue("512m")
+    String spark_kryoserializer_buffer_max();
+
+    @Key("system.spark.driver.maxResultSize")
+    @DefaultValue("0")
+    int spark_driver_maxResultSize();
+
+    @Key("system.spark.driver.userClassPathFirst")
+    @DefaultValue("true")
+    boolean spark_driver_userClassPathFirst();
+
+    @Key("system.spark.io.compression.codec")
+    @DefaultValue("lzf")
+    String spark_io_compression_codec();
+
+    @Key("system.spark.yarn.executor.memoryOverhead")
+    @DefaultValue("600")
+    int spark_yarn_executor_memoryOverhead();
+
+    @Key("system.spark.driver.extraJavaOptions")
+    @DefaultValue("")
+    String spark_driver_extraJavaOptions();
+
+    @Key("system.spark.executor.extraJavaOptions")
+    @DefaultValue("")
+    String spark_executor_extraJavaOptions();
+
+    // ----------------------------------------------------------
+    // Other Options:
+    // ----------------------------------------------------------
+
+    @Key("system.snappy.disable")
+    @DefaultValue("true")
+    boolean snappy_disable();
+
+    // =================================================================================================================
+    // =================================================================================================================
+    // System Options:
+    // =================================================================================================================
+    // =================================================================================================================
+    
+    // ----------------------------------------------------------
+    // General Options:
+    // ----------------------------------------------------------
+
+    @DefaultValue("htsjdk.variant,htsjdk.tribble,org.broadinstitute.hellbender.utils.codecs")
```

Comment:
This one seems to be able to handle whitespace just fine based on a short test with some debug logging:

codec_packages = htsjdk.variant      ,            htsjdk.tribble         ,         org.broadinstitute.hellbender.utils.codecs          
Input:
codec_packages = [htsjdk.variant, htsjdk.tribble, org.broadinstitute.hellbender.utils.codecs]

Logged output: